src/tasks/copy-template-files.ts

```diff
@@ -16,64 +16,21 @@ const EXTERNAL_EXTENSION_TMP_FOLDER = "tmp-external-extension";
 const copy = promisify(ncp);
 let copyOrLink = copy;
 
-const expandExtensions = (options: Options): Extension[] => {
-  const expandedExtensions = options.extensions
-    .map(extension => extensionDict[extension])
-    .map(extDescriptor => [extDescriptor.extends, extDescriptor.value].filter(isDefined))
-    .flat()
-    // this reduce just removes duplications
-    .reduce((exts, ext) => (exts.includes(ext) ? exts : [...exts, ext]), [] as Extension[]);
-
-  return expandedExtensions;
-};
-
 const isTemplateRegex = /([^/\\]*?)\.template\./;
 const isPackageJsonRegex = /package\.json/;
-const isYarnLockRegex = /yarn\.lock/;
-const isNextGeneratedRegex = /packages\/nextjs\/generated/;
 const isConfigRegex = /([^/\\]*?)\\config\.json/;
 const isArgsRegex = /([^/\\]*?)\.args\./;
 const isExtensionFolderRegex = /extensions$/;
 const isPackagesFolderRegex = /packages$/;
 
-const copyBaseFiles = async ({ dev: isDev }: Options, basePath: string, targetDir: string) => {
+const copyBaseFiles = async (basePath: string, targetDir: string) => {
   await copyOrLink(basePath, targetDir, {
     clobber: false,
     filter: fileName => {
-      // NOTE: filter IN
       const isTemplate = isTemplateRegex.test(fileName);
-      const isPackageJson = isPackageJsonRegex.test(fileName);
-      const isYarnLock = isYarnLockRegex.test(fileName);
-      const isNextGenerated = isNextGeneratedRegex.test(fileName);
-
-      const skipAlways = isTemplate || isPackageJson;
-      const skipDevOnly = isYarnLock || isNextGenerated;
-      const shouldSkip = skipAlways || (isDev && skipDevOnly);
-
```

Comment:
Trying to understand this.

- why were we not copying package.json before?
- `dev` is not important anymore here?

Reply:
Ohh my bad actually I planned to write a comment on why did I remove all this and  forgot about it but here is description: 

So basically this function copies ["base"](https://github.com/scaffold-eth/create-eth/tree/main/templates/base) dir into user mentioned path. 

> why were we not copying package.json before?

We were ignore package.json here because just below this `copyOrLink` func we were using `mergePackageJson` to merge package.json's : 

https://github.com/scaffold-eth/create-eth/blob/6a3afc72e7265a4f0d8f4c834f56c252ca41b12c/src/tasks/copy-template-files.ts#L57-L62

I have removed this part in this PR since: 

1. "base" dir is the first dir copied to user mentioned path. 
2. which means the user mentioned path is empty. 
3. So we don't need merging of package.json at all. we could just copy the whole "base" dir to user mentioned path directly except template files. (hence we are not skipping package.json in `copyOrLink` func in this PR)

Reply:
> `dev` is not important anymore here?

I initially thought it was not needed since we have removed "nextjs/generated"  but just updated the logic and added it back so that we don't symlink `yarn.lock` and `packages/nextjs/contracts/deployedContracts.ts` since both are reproducible and we don't want to commit them while creating extension.